lib/susedistribution.pm

```diff
@@ -256,34 +256,40 @@ sub x11_start_program {
     $args{match_no_wait} //= 0;
     $args{match_timeout} //= 90 if check_var('DESKTOP', 'kde');
 
-    # Start desktop runner and type command there
-    init_desktop_runner($program, $timeout);
-    # With match_typed we check typed text and if doesn't match - retrying
-    # Is required by firefox test on kde, as typing fails on KDE desktop runnner sometimes
-    if ($args{match_typed} && !check_screen($args{match_typed}, 30)) {
+    for my $retry (0 .. 2) {
+        # close potential overlays
         send_key 'esc';
+        # Start desktop runner and type command there
         init_desktop_runner($program, $timeout);
+        # With match_typed we check typed text and if doesn't match - retrying
+        # Is required by firefox test on kde, as typing fails on KDE desktop runnner sometimes
+        if ($args{match_typed} && !check_screen($args{match_typed}, 30)) {
+            send_key 'esc';
+            init_desktop_runner($program, $timeout);
+        }
+        wait_still_screen(3);
+        save_screenshot;
+        send_key 'ret';
+        # As above especially krunner seems to take some time before disappearing
+        # after 'ret' press we should wait in this case nevertheless
+        wait_still_screen(3, similarity_level => 45) unless ($args{no_wait} || ($args{valid} && $args{target_match} && !check_var('DESKTOP', 'kde')));
+        return unless $args{valid};
+        set_var('IN_X11_START_PROGRAM', $program);
+        my @target = ref $args{target_match} eq 'ARRAY' ? @{$args{target_match}} : $args{target_match};
+        for (1 .. 3) {
+            push @target, check_var('DESKTOP', 'kde') ? 'desktop-runner-plasma-suggestions' : 'desktop-runner-border';
+            assert_screen([@target], $args{match_timeout}, no_wait => $args{match_no_wait});
+            last unless match_has_tag('desktop-runner-border') || match_has_tag('desktop-runner-plasma-suggestions');
+            wait_screen_change {
+                send_key 'ret';
+            };
+        }
+        set_var('IN_X11_START_PROGRAM', undef);
+        # asserting program came up properly
+        last                                                                if match_has_tag @target;
+        next                                                                if match_has_tag qw(desktop-runner-border desktop-runner-plasma-suggestions);
```

Comment:
@DrMullings Could you tidy up again with this changes? to see how the code looks like? Tidy rules need to be changed... seriously :(

```suggestion
my @runner_tags = qw(desktop-runner-border desktop-runner-plasma-suggestions);
next if match_has_tag @runner_tags;
```